apps/api/src/expenses/expenses.service.ts

```diff
@@ -0,0 +1,52 @@
+import { Injectable, Logger, NotFoundException } from '@nestjs/common'
+import { CreateExpenseDto } from './dto/create-expense.dto'
+import { PrismaService } from '../prisma/prisma.service'
+import { Expense } from '@prisma/client'
+
+@Injectable()
+export class ExpensesService {
+  constructor(private prisma: PrismaService) { }
+
+  async createExpense(createExpenseDto: CreateExpenseDto) {
+    return await this.prisma.expense.create({ data: createExpenseDto })
+  }
+
+  async listExpenses(): Promise<Expense[]> {
+    return this.prisma.expense.findMany({ where: { deleted: false } })
+  }
+
+  async findOne(id: string) {
+    try {
+      const expense = await this.prisma.expense.findFirst({ where: { id, deleted: false } })
+      // handle what happens if it finds the id but the deleted is true , what do we return to the client?
+      // may be a message saying the expense has been deleted ?
+      return expense
+    } catch (error) {
+      throw new NotFoundException('No expense found with that id.')
+    }
+  }
```

Comment:
for returning the deleted expenses we can add another function parameter returnDeleted = false and use it in the where clause. This way the user on the frontend can request the deleted expenses if interested in them too.